src/shared/cache/PmidCache.ts

```diff
@@ -1,53 +1,39 @@
-import {default as SimpleCache, ICache} from "shared/lib/SimpleCache";
+import LazyMobXCache from "../lib/LazyMobXCache";
+import {AugmentedData} from "../lib/LazyMobXCache";
 
-export default class PmidCache extends SimpleCache<any, number[]>
+async function fetch(pmids: number[])
 {
-    constructor()
-    {
-        super();
-    }
-
-    protected async fetch(pmids: number[])
-    {
-        const cache: ICache<any> = {};
-
-        try {
-            const pmidData:any = await new Promise((resolve, reject) => {
-                // TODO better to separate this call to a configurable client
-                $.post(
-                    'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&retmode=json',
-                    {id: pmids.join(',')}
-                ).then((data) => {
-                    resolve(data);
-                }).fail((err) => {
-                    reject(err);
-                });
+        const pmidData:any = await new Promise((resolve, reject) => {
+            // TODO better to separate this call to a configurable client
+            $.post(
+                'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&retmode=json',
+                {id: pmids.join(',')}
+            ).then((data) => {
+                resolve(data);
+            }).fail((err) => {
+                reject(err);
             });
+        });
 
-            pmids.forEach((pmid:number) => {
-                if (pmidData.result[pmid]) {
-                    cache[pmid.toString()] = {
-                        status: "complete",
-                        data: pmidData.result[pmid]
-                    };
-                }
-                else {
-                    cache[pmid.toString()] = {
-                        status: "complete"
-                    };
-                }
+        const ret:AugmentedData<any, string>[] = [];
+        for (const pmidStr of Object.keys(pmidData)) {
+            ret.push({
+                data: pmidData[pmidStr],
+                meta: pmidStr
             });
-
-            this.putData(cache);
         }
-        catch (err) {
-            pmids.forEach((pmid:number) => {
-                cache[pmid.toString()] = {
-                    status: "error"
-                };
-            });
 
-            this.putData(cache);
-        }
+        return ret;
+}
+
+export default class PmidCache extends LazyMobXCache<number, any, string>
```

Comment:
yes, that's the whole point of this pull request :) in typescript 2.3 we can have default generic types but not in typescript 2. Maybe it's time to upgrade to 2.3 for this PR? because that really is quite inconvenient